livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LivestreamComposer.kt

```diff
@@ -1,78 +1,114 @@
 @Composable
 fun LivestreamComposer(
     modifier: Modifier = Modifier,
+    channelId: String,
     composerViewModel: MessageComposerViewModel
 ) {
+    val context = LocalContext.current
     Row(
         modifier
-            .fillMaxWidth(),
+            .fillMaxWidth()
+            .padding(start = 8.dp, end = 8.dp, bottom = 8.dp),
         verticalAlignment = Alignment.CenterVertically
     ) {
         MessageInput(
             modifier = Modifier
                 .fillMaxWidth()
-                .padding(start = 8.dp)
-                .weight(1f),
+                .weight(1f)
+                .padding(start = 8.dp),
             value = composerViewModel.input,
-            attachments = emptyList(),
+            attachments = composerViewModel.selectedAttachments,
             activeAction = composerViewModel.activeAction,
             onValueChange = { composerViewModel.setMessageInput(it) },
-            onAttachmentRemoved = {},
+            onAttachmentRemoved = { composerViewModel.removeSelectedAttachment(it) },
             label = {
-                Row(
-                    Modifier.wrapContentWidth(),
-                    verticalAlignment = Alignment.CenterVertically
-                ) {
-                    Text(
-                        modifier = Modifier.padding(start = 4.dp),
-                        text = "Add your comments here",
-                        color = ChatTheme.colors.textLowEmphasis
-                    )
-                }
-            }
-        )
-        val isInputValid = composerViewModel.input.isNotEmpty()
-        IconButton(
-            enabled = isInputValid,
-            content = {
-                Icon(
-                    imageVector = Icons.Default.Send,
-                    contentDescription = stringResource(id = R.string.stream_compose_send_message),
-                    tint = if (isInputValid) ChatTheme.colors.primaryAccent else ChatTheme.colors.textLowEmphasis
+                Text(
+                    modifier = Modifier
+                        .padding(start = 4.dp)
+                        .wrapContentWidth()
+                        .align(Alignment.CenterVertically),
+                    text = stringResource(id = R.string.composer_hint),
+                    color = ChatTheme.colors.textLowEmphasis
                 )
-            },
-            onClick = {
-                if (isInputValid) {
-                    composerViewModel.sendMessage(composerViewModel.input)
-                }
             }
         )
+        Column(
+            modifier = Modifier
+                .padding(start = 4.dp)
+                .clip(RoundedCornerShape(8.dp))
+                .background(context.cardBackground())
+                .width(40.dp)
+                .height(40.dp)
+                .clickable(
+                    interactionSource = remember { MutableInteractionSource() },
+                    indication = rememberRipple(bounded = false),
+                    onClick = {
```

Comment:
We're missing checks for the input being empty. Right now the user can send empty messages and just keep spamming this button.

We should add an enabled/disabled state here for the icon color and the button.

Reply:
Updated , missed on my part when i converted to custom design instead of using existing bound components